tests/CompileAndExec.integ_test.cpp

```
@@ -0,0 +1,817 @@
+	std::ostringstream output{};
+	std::streambuf* oldBuf = std::cout.rdbuf();
+	std::cout.rdbuf(output.rdbuf());
+
+	// Execution
+	PROFILE_EXEC_START
+	{
+		VirtualMachine vm(pool, registry, reinterpret_cast<AsTL::OpaqueExecCtx*>(&mockSimCtx));
+		vm.setProgram(rawInstructions);
+
+		while (exitCode != Compiler::VMExitCode::SUCCESS) {
+			if (exitCode == Compiler::VMExitCode::FINISHED_EXEC_TICK)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)